### PR TITLE
Initial refactoring of the project

### PR DESCRIPTION
**Pull request by harshanavkis**
Split the project into separate files.

### CODE REVIEW

File: src/vhost_user_vsock_def_types.rs [hunk 1]
```diff
@@ -0,0 +1,6 @@
+use crate::vhost_user_vsock_err::Error;
+use vhost::vhost_user::Error as VhostUserError;
+
+pub type VhostUserResult<T> = std::result::Result<T, VhostUserError>;
```
Comment (stefano-garzarella) [inline src/vhost_user_vsock_def_types.rs]: It seems that we never use VhostUserResult:

```bash
$ cargo clippy
warning: type alias is never used: `VhostUserResult`
 --> src/vhost_user_vsock_def_types.rs:4:1
  |
4 | pub type VhostUserResult<T> = std::result::Result<T, VhostUserError>;
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  |
  = note: `#[warn(dead_code)]` on by default

warning: 1 warning emitted

    Finished dev [unoptimized + debuginfo] target(s) in 0.05s
```

**Review by stefano-garzarella — CHANGES_REQUESTED** [review]
`cargo clippy` complains about `VhostUserResult`

In addition, I think we can remove `vhost_user_vsock_` prefix from each files.
And we can merge def_types, const, and err files in a single file, or even better include them in lib.rs, then we can use Error for example with `use super::Error` in the other files.

**Comment by harshanavkis** [comment]
@stefano-garzarella I have made the requested changes. However, clippy does not complain now about the unused type alias.

**Comment by stefano-garzarella** [comment]
@harshanavkis thanks!
This PR LGTM!

Just a couple of suggestions for next steps:

1. I suggest to fix your git author setup with "Name Surname", since most of upstream projects requires it. For this repository is fine, since for now it is only a PoC, but when we will push on rust-vmm crates, kata, etc. they may have this requirement. For example the Linux kernel doc explains it very well [here](https://www.kernel.org/doc/html/v4.17/process/submitting-patches.html#sign-your-work-the-developer-s-certificate-of-origin)
2. `VhostUserResult` seems really unused, so I think we can clean a bit removing it in next patches/PRs

**Review by stefano-garzarella — APPROVED** [review]

**Pull request merged**